prime-router/src/main/kotlin/credentials/CredentialService.kt

```diff
@@ -0,0 +1,36 @@
+package gov.cdc.prime.router.credentials
+
+import org.apache.logging.log4j.kotlin.Logging
+
+abstract class CredentialService : Logging {
+
+    companion object {
+        private val URL_SAFE_KEY_PATTERN = Regex("^[a-zA-Z0-9_-]*$")
+    }
+
+    /* Methods to implement in subclasses */
+
+    protected abstract fun fetchCredential(connectionId: String): Credential?
+    protected abstract fun saveCredential(connectionId: String, credential: Credential)
+
+    /* Base implementation for credentialService with validations */
+
+    fun fetchCredential(connectionId: String, callerId: String, reason: CredentialRequestReason): Credential? {
+        require(URL_SAFE_KEY_PATTERN.matches(connectionId)) { "connectionId must match: ${URL_SAFE_KEY_PATTERN.pattern}" }
```

Comment:
Is the callerId expected to be? Suggest a comment block would be helpful.

Reply:
I didn't have a strong option on a callerId, and I wasn't sure if we'd end with a UUID on the settings API. Right now it's using the same format as the SftpTransport which is expecting `org_service` ala `AZ_PHD__ELR`.

@RickHawesUSDS With the settings API, it looks like you're sticking with organization name, so in that case the expected value will remain the same. If that's correct I can update the method docs to make that clear. Thoughts?